### ISSUE TITLE
ngrx9 - setResult fail

### ISSUE THREAD

**Issue by pburkindine** [ngrx/platform#2474]
ref failed repro attempt: 
https://github.com/pburkindine/ngrx9-mockselectors

We are unable to upgrade to ngrx 9 because our ngOnDestroy tests for selector streams in Angular components all fail with:

```
 Expected one assertion to be called but received zero assertion calls.

      103 |       storeMock.refreshState();
      104 | 
    > 105 |       expect.assertions(1);
          |              ^
      106 |     }));
      107 |   });
```

Important Note - The test passes when run in isolation. 

I think the problem relates to the value already having been set by the previous test and not cleared successfully, as though possibly the storeMock instance were shared or not cleaned up...

...because if I remove the skip(1) in the destroy test it passes, which it shouldn't because the first value should be null.  (The only other place that value could have been set onto the stream is the previous "consumes values" test.)

## Minimal reproduction of the bug/regression with instructions:

https://github.com/pburkindine/ngrx9-mockselectors

Note that this repro does NOT demonstrate the behavior, but is in every important way identical to the code demonstrating the behavior (a simple string selector in a component, with a `consumes` and an `ngOnDestroy` test in the spec, same nx versions & config etc)

## Expected behavior:

ngOnDestroy test passes (as in the repro)

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):

ngrx: 9.0.0
Angular: 9.1
node: 12.13.1
test env: jest 25.2.4

**Comment by timdeschryver** [comment]
AFAIK the selectors [should be reset](https://github.com/ngrx/platform/blob/master/modules/store/testing/src/mock_store.ts#L18), and there are also some tests that verify this. 

A reproduction here would be useful, as I'm unsure where to start looking.

Have you tried using selectors or function selectors in the real code base, or string selector in the reproduction?

**Comment by pburkindine** [comment]
```
Have you tried using selectors or function selectors in the real code base, or string selector in the reproduction?
```

I'm not following the question, sorry?

**Comment by timdeschryver** [comment]
In the issue you're mentioning that you've used string selectors, but in the reproduction I see that a real selector is being used.

**Comment by pburkindine** [comment]
@timdeschryver 

> In the issue you're mentioning that you've used string selectors, but in the reproduction I see that a real selector is being used.

Sorry, I mean that the selector is selecting a simple string from state (as in, simplest possible case, not an object or anything complicated)

**Comment by pburkindine** [comment]
@timdeschryver the code you pointed me to finally got clear what was happening.  We had extended MockStore when v8 first came out so that we could implement a `fireSelector` function (overrideSelector, setState) (see https://github.com/ngrx/platform/issues/2000)).  The problem was that the code you pointed to was trying to pull the base MockStore from state, finding it missing, and so not resetting the selectors.  Since .refreshState is available now, all I needed to do was destroy our extended class.  Thanks for pointing me in the right direction, much obliged!
